src/test/java/com/snowflake/kafka/connector/internal/IngestSdkProviderTest.java

```diff
@@ -0,0 +1,88 @@
+package com.snowflake.kafka.connector.internal;
+
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+import com.snowflake.kafka.connector.Utils;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Map;
+
+public class IngestSdkProviderTest {
+    private Map<String, String> config;
+    private String connectorName;
+
+    @Before
+    // set up sunny day tests
+    public void setup() {
+        this.connectorName = "testConnector";
+
+        // config
+        this.config = TestUtils.getConfForStreaming();
+        SnowflakeSinkConnectorConfig.setDefaultValues(this.config);
+    }
+
+    @Test
+    public void testCreateClient() {
+        // setup
+        IngestSdkProvider ingestSdkProvider = new IngestSdkProvider();
+        SnowflakeStreamingIngestClient goalClient = TestUtils.createStreamingClient(this.config, "KC_CLIENT_" + this.connectorName + "0");
+
+        // test
+        SnowflakeStreamingIngestClient createdClient = ingestSdkProvider.createStreamingClient(this.config, this.connectorName);
+
+        // verification - very difficult (impossible?) to mock the SnowflakeStreamingIngestClientFactory.builder method because it is static, so use a goal client to at least test idempotency
```

Comment:
Had trouble mocking a static factory, open to suggestions on how to test it. Ultimately I'm leaning towards changing the factory but that is out of scope for this